src/API/Site/Controllers/Google/AccountController.php

```diff
@@ -83,17 +85,38 @@ public function register_routes(): void {
 	 * @return callable
 	 */
 	protected function get_connect_callback(): callable {
-		return function() {
+		return function( Request $request ) {
 			try {
+				$next = $request->get_param( 'next' );
+				$path = $next === 'setup-mc' ? '/google/setup-mc' : '/google/settings&subpath=/reconnect-accounts';
+
 				return [
-					'url' => $this->connection->connect( admin_url( 'admin.php?page=wc-admin&path=/google/setup-mc' ) ),
+					'url' => $this->connection->connect( admin_url( "admin.php?page=wc-admin&path=$path" ) ),
 				];
 			} catch ( Exception $e ) {
 				return new WP_REST_Response( [ 'message' => $e->getMessage() ], $e->getCode() ?: 400 );
```

Comment:
Not related to this PR, but this should be just Response.

```suggestion
				return new Response( [ 'message' => $e->getMessage() ], $e->getCode() ?: 400 );
```